### PR TITLE
feat: update to rules_js 1.27.0 and pickup fixed_args feature

### PR DESCRIPTION
**Pull request by gregmagolan**
rules_js v1.27.0 adds a new fixed_args feature to js_binary & js_test that can be trivially added to jest_test.

---

### Type of change

- New feature or functionality (change which adds functionality)

### Test plan

- New test cases added

### CODE REVIEW

File: jest/private/jest_test.bzl [hunk 1]
```diff
@@ -65,7 +65,10 @@ def _impl(ctx):
     if ctx.attr.chdir:
         unwind_chdir_prefix = "/".join([".."] * len(ctx.attr.chdir.split("/"))) + "/"
 
-    fixed_args = [
+    fixed_args = []
+    if hasattr(ctx.attr, "fixed_args"):
```
Comment (jbedard) [inline jest/private/jest_test.bzl:69]: Is the `hasattr` only required because an older version of `rules_js` might be used?

Comment (gregmagolan) [reply jest/private/jest_test.bzl:69]: Yea, shouldn't happen with bzlmod but it can with someone's poorly written WORKSPACE file.

Comment (jbedard) [reply jest/private/jest_test.bzl:69]: Can we drop it at some point?

Comment (gregmagolan) [reply jest/private/jest_test.bzl:69]: For rules_jest we can with the 1.0 release since its still pre-1.0. I'll add a TODO.

File: MODULE.bazel [hunk 2]
```diff
@@ -6,8 +6,8 @@ module(
     version = "0.0.0",
 )
 
-bazel_dep(name = "aspect_bazel_lib", version = "1.30.2")
-bazel_dep(name = "aspect_rules_js", version = "1.26.0")
+bazel_dep(name = "aspect_bazel_lib", version = "1.32.1")
```
Comment (gregmagolan) [inline MODULE.bazel:9]: Some interaction between the aspect_bazel_lib in root WORKSPACE.bazel and the version here leads to this error:
```
ERROR: /home/runner/.cache/bazel/_bazel_runner/75700d67e261d1f3b82aa7f70f76b4b9/external/expand_template_toolchains/BUILD.bazel:19:10: no such target '@aspect_bazel_lib~1.31.2//lib:expand_template_toolchain_type': target 'expand_template_toolchain_type' not declared in package 'lib' defined by /home/runner/.cache/bazel/_bazel_runner/75700d67e261d1f3b82aa7f70f76b4b9/external/aspect_bazel_lib~1.31.2/lib/BUILD.bazel (Tip: use `query "@aspect_bazel_lib//lib:*"` to see all the targets in that package) and referenced by '@expand_template_toolchains//:darwin_arm64_toolchain'
ERROR: /home/runner/.cache/bazel/_bazel_runner/75700d67e261d1f3b82aa7f70f76b4b9/external/bazel_tools/src/tools/launcher/BUILD:9:14: While resolving toolchains for target @bazel_tools//src/tools/launcher:launcher: invalid registered toolchain '@expand_template_toolchains//:darwin_arm64_toolchain': Analysis failed
```
if we also don't bump the version here.

Erring on the side of bumping this version so the expand_template toolchain is available for users.

File: jest/private/jest_test.bzl (hunk 1 above)
Comment (jbedard) [reply jest/private/jest_test.bzl:71]: Yeah I'd like to add TODOs to all these so we remember to drop it in the future.

**Review by jbedard — APPROVED** [review]

**Pull request merged**